test/Microsoft.OpenApi.Tests/Services/OpenApiValidatorTests.cs

```diff
@@ -48,9 +48,42 @@ public void ResponseMustHaveADescription()
             validator.Errors.ShouldBeEquivalentTo(
                     new List<ValidationError>
                     {
-                        new ValidationError(ErrorReason.Required, "#/description",
+                        new ValidationError(ErrorReason.Required, "#/paths/~1test/get/responses/200/description",
                             String.Format(SRResource.Validation_FieldIsRequired, "description", "response"))
         });
         }
+
+        [Fact]
+        public void ServersShouldBeReferencedByIndex()
+        {
+            var openApiDocument = new OpenApiDocument();
+            openApiDocument.Info = new OpenApiInfo()
+            {
+                Title = "foo",
+                Version = "1.2.2"
+            };
+            openApiDocument.Servers = new List<OpenApiServer> {
+                new OpenApiServer
+                {
+                    Url = "http://example.org"
+                },
+                new OpenApiServer
+                {
+
+                }
+            };
+            
+            var validator = new OpenApiValidator();
+            var walker = new OpenApiWalker(validator);
+            walker.Walk(openApiDocument);
+
+            validator.Errors.ShouldBeEquivalentTo(
+                    new List<ValidationError>
+                    {
+                        new ValidationError(ErrorReason.Required, "#/servers/1/url",
+                            String.Format(SRResource.Validation_FieldIsRequired, "url", "server"))
+        });
+        }
+
     }
 }
```

Comment:
Any value in having a test with custom extensions?

Reply:
I put the validation test into the existing custom extension reading test https://github.com/Microsoft/OpenAPI.NET/pull/176/files#diff-f884e65f4ced1eff8c8cfc2496d317d1R33  I'll split out the tests to put stuff in the right place.

Reply:
Done.